staging/src/k8s.io/client-go/util/workqueue/delaying_queue.go

```diff
@@ -286,6 +286,7 @@ func (q *delayingType[T]) waitingLoop() {
 			}
 
 			entry = heap.Pop(waitingForQueue).(*waitFor)
+			q.metrics.removeFromWaitingForQueue()
```

Comment:
can not you update the. metrics with the value of `waitingForQueue.Len()` directly?

Reply:
👍 Changed the metric to be a `SettableGaugeMetric` which allows us to set the length directly.